jaxrs-api/src/main/java/jakarta/ws/rs/client/FactoryFinder.java

```diff
@@ -168,4 +163,44 @@ private static Object newInstance(final String className, final ClassLoader clas
         throw new ClassNotFoundException(
                 "Provider for " + factoryId + " cannot be found", null);
     }
+
+    private static ClassLoader getClassLoader() {
+        if (System.getSecurityManager() == null) {
+            return FactoryFinder.class.getClassLoader();
+        }
+        return AccessController.doPrivileged((PrivilegedAction<ClassLoader>) FactoryFinder.class::getClassLoader);
+    }
+
+    private static <T> T findFirstService(final String factoryId, final ClassLoader cl, final Class<T> service) {
+        final FindServiceAction<T> action = new FindServiceAction<>(factoryId, cl, service);
+        if (System.getSecurityManager() == null) {
+            return action.run();
+        }
+        return AccessController.doPrivileged(action);
```

Comment:
I understand putting the `getClassLoader` calls into a doPriv block, but I don't understand the `ServiceLoader` calls.  Does that need to be in a doPriv?  The javadoc doesn't mention that it throws a `SecurityException`.

Reply:
The `ServiceLoader.load()` is actually safe, but it captures the `AccessContrelContext` is captured and during iteration a security exception could be thrown on `Constructor.newInstance()`.

From the JavaDocs:

> Service loaders always execute in the security context of the caller. Trusted system code should typically invoke the methods in this class, and the methods of the iterators which they return, from within a privileged security context.